sdk/core/Azure.Core.Experimental/src/MutableJsonDocument.cs

```diff
@@ -56,15 +56,15 @@ public void WriteTo(Stream stream, StandardFormat format = default)
                 throw new ArgumentOutOfRangeException(nameof(format));
             }
 
-            Utf8JsonWriter writer = new Utf8JsonWriter(stream);
             if (!Changes.HasChanges)
             {
                 Write(stream, _original.Span);
                 stream.Flush();
                 return;
             }
 
-            WriteRootElementTo(writer);
+            Utf8JsonWriter writer = new(stream);
+            RootElement.WriteTo(writer);
```

Comment:
Does this automatically flush?   If not, should we consider doing so in order to keep behavior consistent with the `!Changes.HasChanges` path?

Reply:
Now that `using` was added it will.

Reply:
It won't also flush the stream though, right?  I'll have a quick look at whether JsonElement.WriteTo() flushes to make sure we have parity in behavior here.

Reply:
Not sure. The documentation doesn't say that, only that it flushes any remaining JSON scopes that may have been buffered.

Reply:
Generally, though, you should always dispose a disposable. I'm surprise Roslyn Analyzers didn't flag that missing usage. I know I've seen those warnings before in our repo. Wonder if it was locally disabled e.g., for this project...and why.

Reply:
Yep, good call.  I know there are some differences in what's enforced in Experimental vs. Core, so I'm not too surprised.

Reply:
It looks like JsonDocument/JsonElement does not flush the writer themselves in WriteTo(), so I've removed places I'm doing that.  I think it makes sense that a perf-minded caller would want to own this themselves.